src/fetchers/git-fetcher.js

```diff
@@ -73,7 +73,17 @@ export default class GitFetcher extends BaseFetcher {
     await git.initRemote();
     await git.clone(this.dest);
 
-    let tarballInMirrorPath = this.config.getOfflineMirrorPath(this.reference);
+    // Get the tarball filename from the url
+    const {pathname} = url.parse(this.reference);
+    let tarballFilename;
+    if (pathname == null) {
+      tarballFilename = this.reference;
+    } else {
+      tarballFilename = path.basename(pathname);
+    }
+
+    let tarballInMirrorPath = this.config.getOfflineMirrorPath(tarballFilename);
```

Comment:
@bestander I'm not sure if a non-breaking approach is possible without adding a lot of special logic for backwards compatibility, but this could be OK. My understanding is that the breaking change affects only people who use an offline mirror and have a project that depends on a scoped package. Some of those people also have buggy lockfiles that sometimes trigger "bad hash" errors. After this PR they will get just one more "bad hash" error that will go away after they regenerate the lockfile.